lib/galaxy/tools/parameters/basic.py

```diff
@@ -344,9 +358,8 @@ def from_html( self, value, trans=None, other_values={} ):
         try:
             return int( value )
         except:
-            if isinstance( value, basestring ):
-                if value.startswith( "$" ) and ( trans is None or trans.workflow_building_mode ):
-                    return value
+            if contains_workflow_parameter(value) and _allow_workflow_parameters_in_context(trans):
+                return value
             elif not value and self.optional:
                 return ""
             if trans is None or trans.workflow_building_mode:
```

Comment:
Also here change `trans is None or trans.workflow_building_mode` to `_allow_workflow_parameters_in_context(trans)`